lib/code-signing/private-key.ts

```diff
@@ -63,28 +63,27 @@ export class RsaPrivateKeySecret extends cdk.Construct {
 
     props.deletionPolicy = props.deletionPolicy || cdk.DeletionPolicy.Retain;
 
+    const codeLocation = path.join(__dirname, 'private-key');
     const customResource = new lambda.SingletonFunction(this, 'ResourceHandler', {
       uuid: '72FD327D-3813-4632-9340-28EC437AA486',
       description: 'Generates an RSA Private Key and stores it in AWS Secrets Manager',
       runtime: lambda.Runtime.Python36,
       handler: 'index.main',
-      code: new lambda.InlineCode(
-        fs.readFileSync(path.join(__dirname, 'private-key.py'))
-          .toString('utf8')
-          // Remove blank and comment-only lines, to shrink code length
-          .replace(/^[ \t]*(#[^\n]*)?\n/mg, '')
-      ),
+      code: new lambda.AssetCode(codeLocation),
       timeout: 300,
     });
 
     this.secretArnLike = cdk.ArnUtils.fromComponents({
       service: 'secretsmanager',
       resource: 'secret',
       sep: ':',
-      resourceName: `${props.secretName}-*`
+      resourceName: `${props.secretName}-??????`
```

Comment:
What's this about? Does it actually work?

Reply:
In fact this is what the AWS documentation for SecretsManager recommends. The `?` in ARN patterns matches exactly one character, and secret arms are composed with the name to which `-` and 6 random characters are added. That is the "safe" way to grant permissions to a secret before it is created (or if you can't pass around the actual full ARN). It has been tested to work.